crates/noirc_evaluator/src/ssa/acir_gen/operations/cmp.rs

```diff
@@ -64,28 +69,30 @@ pub(crate) fn evaluate_neq(
 
         return Expression::from(&constraints::evaluate_zero_equality(x_witness, evaluator));
     }
+
+    // Arriving here means that `lhs` and `rhs` are not Arrays
     let l_c = l_c.expect("ICE: unexpected array pointer");
     let r_c = r_c.expect("ICE: unexpected array pointer");
```

Comment:
I'm guessing here, we don't need to check if `l_c == r_c` ?

Reply:
No we don't, that's why I removed it. If they are equal, then x will be 0.

Reply:
Makes sense ser, merging